src/poetry/installation/wheel_installer.py

```diff
@@ -66,12 +87,52 @@ def for_source(self, source: WheelFile) -> WheelDestination:
         scheme_dict["headers"] = str(Path(scheme_dict["headers"]) / source.distribution)
 
         return self.__class__(
-            scheme_dict,
+            source=source,
+            scheme_dict=scheme_dict,
             interpreter=self.interpreter,
             script_kind=self.script_kind,
             bytecode_optimization_levels=self.bytecode_optimization_levels,
         )
 
+    def _validate_hash_and_size(
+        self, records: Iterable[tuple[Scheme, RecordEntry]]
+    ) -> None:
+        record_lines = self._source.read_dist_info("RECORD").splitlines()
+        record_mapping = {
+            record[0]: record for record in parse_record_file(record_lines)
+        }
+        for item in self._source._zipfile.infolist():
+            record_args = record_mapping.pop(item.filename, None)
+            if not record_args:
+                continue
+
+            file_record = RecordEntry.from_elements(*record_args)
+            computed_record = next(
+                record for _, record in records if record.path == item.filename
+            )
```

Comment:
This doesn't always work. I tested a large project and got a `StopIteration` for some packages.

You can reproduce it by running `poetry add fonttools` for example.

What's even worth: The package is installed, but the `RECORD` file is not written.

`poetry run pip uninstall fonttools` now throws `ERROR: Cannot uninstall fonttools 4.39.4, RECORD file not found. Hint: The package was installed by Poetry 1.6.0.dev0.` and `poetry run pip install fonttools` gives `Requirement already satisfied`.

You can only "manually" uninstall the package.